charts/opentelemetry-operator/values.yaml

```diff
@@ -85,7 +85,7 @@ kubeRBACProxy:
       memory: 64Mi
 
 ## Admission webhooks make sure only requests with correctly formatted rules will get into the Operator.
-##
+## They also enable the sidecar injection for OpenTelemetryCollector and Instrumentation CR's
 admissionWebhooks:
   failurePolicy: Fail
   enabled: true
```

Comment:
I think using `create` bool flag would be the most common approach for such use cases instead of adding `customAdmissionWebhooks`. See https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-collector/values.yaml#L137 as an example.

I believe we can deprecate `enabled` in favor of `create` and `secretName` options.

Reply:
@dmitryax I think this covers the 4 possible scenarios:
- using provided web-hooks + cert: needs `admissionWebhooks.create: true`
- using provided web-hooks but a different cert name: needs `admissionWebhooks.create: true` and `admissionWebhooks.secretName: "name"`
- not using web-hooks + cert at all: needs admissionWebhooks.create: false
- using custom sideloaded web-hooks + cert: needs admissionWebhooks.create: false(so no webhooks are created by the chart) + admissionWebhooks.secretName: "name"

WDYT

Reply:
Sounds good to me.

> not using web-hooks + cert at all: needs admissionWebhooks.create: false and setting env var to ENABLE_WEBHOOKS: "false"

Do we need setting env var in that case? Can't `admissionWebhooks.create: false` and `admissionWebhooks.secretName: ""` mean that?

Reply:
> Sounds good to me.
> 
> > not using web-hooks + cert at all: needs admissionWebhooks.create: false and setting env var to ENABLE_WEBHOOKS: "false"
> 
> Do we need setting env var in that case? Can't `admissionWebhooks.create: false` and `admissionWebhooks.secretName: ""` mean that?

the `ENABLE_WEBHOOKS` is true by default and if I remember correctly the manager pod will throw an error.

Reply:
@dmitryax , just tested, the manager pod does not start with that env var set to `false` when AW and cert are missing

Reply:
I mean why do we need to ask user to set the ver? We can set to flase in the template based if `admissionWebhooks.create: false` and `admissionWebhooks.secretName: ""`, right?

Reply:
@dmitryax imo we can't set it based on only these 2 flags and encompass the 4 options outlined above. I believe that this setting will be used by a very small % of our users and caters to more advanced(or that work in more complex envs) users so I would rather have it set explicit rather than implicit.

Reply:
SGTM . in that case we need to add `ENABLE_WEBHOOKS: "true"` into the default config to keep existing behavior, right?

Reply:
Another thing. If you want to make a breaking change, please document the Upgrade process in a file similar to https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-collector/UPGRADING.md